burn-ndarray/src/ops/tensor.rs

```diff
@@ -51,10 +58,16 @@ impl<E: NdArrayElement> TensorOps<NdArrayBackend<E>> for NdArrayBackend<E> {
         distribution: Distribution<E>,
         device: &NdArrayDevice,
     ) -> NdArrayTensor<E, D> {
-        let mut seed = SEED.lock().unwrap();
-        let mut rng: StdRng = match seed.as_ref() {
-            Some(rng) => rng.clone(),
-            None => StdRng::from_entropy(),
+        let mut seed = SEED.lock();
+        let mut rng: SmallRng;
+        rng = if let Some(rng_seeded) = seed.as_ref() {
+            rng_seeded.clone()
+        } else {
+            if cfg!(feature = "build-generated-seed") || cfg!(test) {
+                SmallRng::seed_from_u64(GENERATED_SEED)
+            } else {
+                panic!("Seeding from the OS automatically is not supported.")
+            }
         };
```

Comment:
That's a reason why a `std` feature flag can be usefull. I would automatically generate a seed at compile time when the `no_std` flag is activated, which would avoid having to `panic` here.

Reply:
We can do this for std but we should keep panic if Backend is not seeded (more explicit). The user will have an option to use `build-generated-seed` if needed.

Reply:
What is the use case when you want `no_std` and build without a generated seed?

Reply:
OK. To make consistent with `std` fallback behavior, i.e. get seed automatically, no_std should get from somewhere, which will be from the build time seed. I'll remove the `build-generated-seed` entirely.